### ISSUE TITLE
Allow setting a yearly subscription for promoted add-ons

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-server#15872]
Similar to https://github.com/mozilla/addons-server/issues/15857, we would like the option in the admin to set up a yearly subscription instead of a monthly subscription. By default I think we'll have a yearly subscription for Sponsored and monthly for Verified. This also means that we'll have 4 default rates (Sponsored Monthly, Sponsored Yearly, Verified Monthly, Verified Yearly). Like with https://github.com/mozilla/addons-server/issues/15857, this only determines what we should during onboarding.

**Comment by willdurand** [comment]
> Similar to #15857, we would like the option in the admin to set up a yearly subscription instead of a monthly subscription. By default I think we'll have a yearly subscription for Sponsored and monthly for Verified. This also means that we'll have 4 default rates (Sponsored Monthly, Sponsored Yearly, Verified Monthly, Verified Yearly). Like with #15857, this only determines what we should during onboarding.

mmm, so maybe we should create the following products and prices in Stripe:

- Product: Verified, Price period: monthly
- Product: Sponsored, Price period: yearly

We'll have a way to override the price amount in #15857 and we could add an `onboarding_period` field to override the billing period. What would be the value in having 4 default rates (i.e. prices in Stripe)?

**Comment by willdurand** [comment]
ha ok, so far we support the custom rate and I'll update some env var values to implement:

> By default, making an add-on Sponsored should onboard the developer with a yearly subscription, while Verified should onboard with a monthly subscription.

That being said, we're using the custom rate field as an optional field, i.e. if there is no value, we use the default in Stripe but it won't be exposed in the django admin. Of course that was done before this new requirement.

In my previous comment, I was thinking that we could create an optional `onboarding_period` field as well, which would make the `onboarding_rate` field required, but I guess we can have 4 different Stripe prices (2 per Product) and make those two `onboarding_rate` and `onboarding_period` fields optional and independent.

That way, an admin would be able to express:

- default sponsored sub (yearly): admin does not change anything
- default verified sub (monthly): admin does not change anything
- monthly sponsored sub (with default rate): admin selects `onboarding_period: monthly`
- yearly verified sub (with default rate): admin selects `onboarding_period: yearly`
- yearly sponsored sub with custom rate: admin adds a custom rate
- monthly verified sub with custom rate: admin adds a custom rate
- monthly sponsored with custom rate: admin selects `onboarding_period: monthly` and adds a custom rate
- yearly verified with custom rate: admin selects `onboarding_period: yearly` and adds a custom rate

**Comment by jvillalobos** [comment]
Sounds good!

**Comment by ioanarusiczki** [comment]
@willdurand I tested on AMO dev - FF82(Win10): 

- I've tested all the scenarios mentioned above. These are the default monthly and yearly rates:

$60.00 per Year: default sponsored sub (yearly): admin does not change anything
$10.00 per month : default verified sub (monthly): admin does not change anything
$6.00 Per month: monthly sponsored sub (with default rate): admin selects onboarding_period: monthly
$150.00 per year: yearly verified sub (with default rate): admin selects onboarding_period: yearly

- I did not see issues with the custom rates or onboarding period. 

- when The Onboarding rate or Onboarding period is the default the fields in admin will not show the amount or period of time.

![numbers](https://user-images.githubusercontent.com/33448286/98547124-5fa9b000-22a0-11eb-9a55-5e109e94e3c0.png)

Let me know if I should file an issue.
